File: Agent/.NET 6.0/Authority.cs
namespace Technologai
{
    public class Authority
    {
        public string AuthUri { get; private set; } = "https://auth.technologai.com";        
        public string BrokerUri { get; private set; } = "https://broker.technologai.com";
        public string StreamUri { get; private set; } = "https://stream.technologai.com";
        public string? ClientApi { get; private set; }
        public string? TokenApi { get; private set; }
        
        public Authority(string authUri)
        {
            // TODO: Connect to an OIDC Discovery Endpoint and get the authoratative values

            if (authUri != "https://auth.technologai.com") { throw new NotImplementedException(); }

            this.AuthUri = authUri;            
            this.ClientApi = "/client";
            this.TokenApi = "/token";
        }
    }
}

File: Agent/Python/authority.py
from dataclasses import dataclass

@dataclass
class Authority:
    host: str = "auth.technologai.com"
    client_endpoint: str = None
    token_endpoint: str = None
    broker_host: str = None

    def __init__(self, host):
        # TODO: Connect to Discovery Endpoint and get the correct values
        if host != "auth.technologai.com":
            raise NotImplementedError()

        self.host = host
        self.broker_host = "broker.technologai.com"
        self.client_endpoint = "https://auth.technologai.com/client"
        self.token_endpoint = "https://auth.technologai.com/token"

# Example usage
authority = Authority("auth.technologai.com")
print(authority.host)
print(authority.client_endpoint)
print(authority.token_endpoint)
print(authority.broker_host)
